src/QMCDrivers/GreenFunctionModifiers/DriftModifierUNR.h

```diff
@@ -23,8 +24,9 @@ class DriftModifierUNR : public DriftModifierBase
   using RealType = QMCTraits::RealType;
   using PosType  = QMCTraits::PosType;
 
-  void getDrifts(RealType tau, const std::vector<GradType>& qf, std::vector<PosType>&) const final;
-
+  void getDrifts(RealType tau, const std::vector<GradType>& qf, MCCoords<MCCoordsTypes::RSSPINS>&) const final;
+  void getDrifts(RealType tau, const std::vector<GradType>& qf, MCCoords<MCCoordsTypes::RS>&) const final;
```

Comment:
I like this pattern much more than `conexpr if`. Namely, API becomes reactive. Cody has refactored the ParticleSet APIs
mw_makeMoveWithSpin has been replaced with mw_makeSpinMove.
The old mw_makeMoveWithSpin can be achieved with mw_makeMove and mw_makeSpinMove.
Then I think we can introduce
```
makeMove(MCCoords<MCCoordsTypes::RSSPINS>&)
makeMove(MCCoords<MCCoordsTypes::RS>&)
```
on top of existing APIs.

Reply:
I'd much prefer a templated function but because of the unecessary virtual inheritance in the DriftModifiers we are stuck with the overloading.

Reply:
What you want is a templates function with specialization without the class being templated. We have to go for CRTP I guess. Then do we really want the whole class being templated? Since we anyway need to specialize the function. Just putting distinct functions is totally fine.

Reply:
I think in many cases we don't want the whole class templated. If we do maybe CRTP is useful maybe we actually really gain something from the virtual inheritance.

But drift modifier would be better without inheritance. I think this a case that clearly would be better written with `if constrexpr` and no vtable.  The overloads are going to cost several vtable calls without repeating code.

Reply:
do you mean a `ParticleSet::makeMove` overload set?
That I think makes sense.
Hiding the QMC algorithm in another state machine I'm not in favor of.